packages/engine/src/lib/handler/loop-executor.ts

```diff
@@ -45,13 +46,33 @@ export const loopExecutor: BaseExecutor<LoopOnItemsAction> = {
       path: string;
     };
 
-    const { resolvedInput, censoredInput } =
-      await constants.variableService.resolve<LoopOnActionResolvedSettings>({
-        unresolvedInput: {
-          items: action.settings.items,
-        },
-        executionState,
-      });
+    // eslint-disable-next-line prefer-const
+    const {
+      resolvedInput,
+      censoredInput,
+    }: {
+      resolvedInput: LoopOnActionResolvedSettings;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      censoredInput: any;
+    } = await constants.variableService.resolve<LoopOnActionResolvedSettings>({
+      unresolvedInput: {
+        items: action.settings.items,
+      },
+      executionState,
+    });
+
+    if (isString(resolvedInput.items)) {
```

Comment:
this is ugly as hell but I figured it would be safest to do it here, where we know a valid input must be an array. I think it's better than the behavior today, where it just doesn't return anything, or passes individual chars if you give it a string.